workbench-core/accounts/src/utilities/hostingAccountLifecycleService.ts

```diff
@@ -479,6 +493,17 @@ export default class HostingAccountLifecycleService {
     return currOutputVal;
   }
 
+  private _getEnvironmentFilesPathForArn(artifactBucketArn: string): string {
+    const parsedBucketArn = artifactBucketArn.replace('arn:aws:s3::::', '').split('/');
+    const bucketName = parsedBucketArn[0];
+
+    if (_.isEmpty(bucketName) || bucketName.length === 0) {
+      throw new InvalidArtifactBucketArnError(artifactBucketArn);
```

Comment:
Are we trying to propagate this error message to the requester? If yes, we need to use Boom. You can throw one of the boom errors from [here](https://hapi.dev/module/boom/api/?v=10.0.0#http-4xx-errors).

An example of how Boom is used is shown [here](https://github.com/awslabs/solution-spark-on-aws/blob/8b1fba979da9d3e335cb17d819cba6e2279c654a/workbench-core/accounts/src/services/projectService.ts#L61)

Reply:
> Did you get a chance to test if creating/updating an account through the API still propogated the `environmentInstanceFiles` and `encryptionKeyArn`? `encryptionKeyArn` should be added after the `AccountHandler` runs.

Yup, it all gets propagated to the account.

Reply:
We can just throw a Boom error here, and it'll bubble up to the requester. 
````
console.error(`Cannot calculate artifact bucket name from S3 ARN ${artifactBucketArn}`);
throw Boom.badImplementation('Invalid artifact bucket ARN. Cannot calculate artifact bucket name from S3 ARN); // This message will be shown to the requester. However, we don't want to provide the `artifactBucketArn` because we don't want to leak implementation details to the requester
```